packages/datadog-plugin-rhea/src/index.js

```diff
@@ -107,12 +125,33 @@ function createWrapReceiverDispatch (tracer, config) {
   }
 }
 
-function wrapDeliveryMethod (name, stateName, delivery) {
-  const method = delivery[name]
-  delivery[name] = function (params) {
-    this._dd.span.setTag('amqp.delivery.state', stateName)
-    this._dd.done()
-    return method.apply(this, arguments)
+function getStateFromData (stateData) {
+  if (stateData && stateData.descriptor && stateData.descriptor.value) {
+    switch (stateData.descriptor.value) {
+      case 36: return 'accepted'
+      case 37: return 'rejected'
+      case 38: return 'released'
+      case 39: return 'modified'
+    }
+  }
+}
+
+function wrapDeliveryUpdate (delivery) {
```

Comment:
This could probably be refactored to return the function similar to other `wrapX` functions.

Reply:
`delivery.update` is added to each individual delivery instance manually, so it needs to be patched manually every single time at the moment we first have access to the object. It's not using `this.wrap()` like the `wrapX` function do. Is there another more idiomatic way to do dynamic wrapping like this?

Reply:
I meant doing something like this:

```js
delivery.update = wrapDeliveryUpdate(delivery.update)

function wrapDeliveryUpdate (update) {
  return function updateWithTrace () { ... }
}
```

This would make it similar to the other wrapper functions, and useable as-is if needed with a wrapper factory.